037-auto-rebalance-cluster-scaling.md

```diff
@@ -0,0 +1,177 @@
+# Automatic rebalance on cluster scaling
+
+This feature is about enabling the cluster operator to get an optimization proposal (to be auto or manually approved) and run the corresponding rebalancing when the user scales the cluster up or down by changing the `spec.kafka.replicas` field on the `Kafka` custom resource.
+The scaling operation is then subordinated to the rebalancing operation and if it is approved and runs successfully.
+
+## Current situation
+
+Currently, there is no automatic rebalancing when the user scales the cluster up or down.
+It means that:
+
+* after scaling up the cluster, the user has to create a `KafkaRebalance` custom resource with the `spec.mode: add-brokers` and the `spec.brokers` with the added brokers, in order to rebalance the cluster and moving some replicas to the newly added brokers.
+* before scaling down the cluster, the user has to create a `KafkaRebalance` custom resource with the `spec.mode: remove-brokers` and the `spec.brokers` with the brokers to remove, in order to rebalance the cluster and moving replicas out of the brokers to be removed.
+
+## Motivation
+
+The rebalancing operation during scaling up and down is a manual two step process.
+
+* scaling up the cluster, then running a rebalance.
+* running a rebalance, then scaling down the cluster.
+
+In order to simplify, it would be great for the user to deal with just scaling the cluster by changing the `spec.kafka.replicas` field and then having the operator taking care to run a rebalancing operation automatically if requested.
+
+## Proposal
+
+We would need to extend the `Kafka` custom resource to allow the user specifying that he wants auto-rebalancing on scaling.
+
+A new `spec.kafka.autoRebalance` field could be added for this purpose.
+Because we are talking about auto-rebalancing, the corresponding field could also be not limited to the rebalancing on scaling but a more general “continuous” auto-rebalancing to enable on the cluster. 
+To specify when running the auto-rebalancing, a `spec.kafka.autoRebalance.mode` field could be used by listing if running auto-rebalancing only on scaling, during the normal execution or both.
+Having “continuous” auto-rebalancing is out of scope for this proposal, the field is added for allowing a future implementation.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    replicas: 3
+    # ...
+    autoRebalance:
+      mode: [running, scaling] # or just one of the two
+```
+
+Finally, the user should be able to define what are the [goals and the other rebalancing related configurations](https://strimzi.io/docs/operators/latest/configuring.html#type-KafkaRebalance-reference) which are normally described via a `KafkaRebalance` custom resource.
+In any case, for rebalancing the cluster, an actual `KafkaRebalance` custom resource has to be automatically created by the cluster operator.
```

Comment:
I have updated the proposal in this direction by using different modes of auto-rebalancing (with different corresponding templates) and talking more in general about add and remove brokers instead of scaling up and down.